subsys/mgmt/hawkbit/hawkbit.c

```diff
@@ -358,7 +363,8 @@ static int hawkbit_settings_export(int (*cb)(const char *name, const void *value
 	LOG_DBG("export hawkbit settings");
 	(void)cb("hawkbit/action_id", &hb_cfg.action_id, sizeof(hb_cfg.action_id));
 #ifdef CONFIG_HAWKBIT_SET_SETTINGS_RUNTIME
-	(void)cb("hawkbit/server_addr", &hb_cfg.server_addr, sizeof(hb_cfg.server_addr));
+	(void)cb("hawkbit/server_addr", &hb_cfg.server_addr,
+		 strnlen(hb_cfg.server_addr, sizeof(hb_cfg.server_addr)));
```

Comment:
```suggestion
		 strnlen(hb_cfg.server_addr, sizeof(hb_cfg.server_addr)) + 1);
```

Reply:
Why the +1? I don't think you want to pass a length off the end of the array. The length should be at maximum `sizeof(hb_cfg.server_addr))` if the string is not null terminated.

Reply:
We want to pass the length of the string including zero termination. Maybe we should use the normal strlen here.

Reply:
the string, including the null termination byte, should be within the bounds of the array. I don't understand why you would want strnlen to read off the bounds of `hb_cfg.server_addr`.

Reply:
The + 1 is after the strnlen() not inside.

Reply:
> The + 1 is after the strnlen() not inside.

Okay, but what happens when strnlen returns sizeof(hb_cfg.server_addr)? The cb will be passed `sizeof(hb_cfg.server_addr) + 1`. I'm not very familiar with the settings subsystem, so I'm not sure what is passed as `cb`.

Reply:
It shouldn't, when setting it we already check, that it's not too long. strlen should be fine enough. In the zephyr codebase strnlen is rarely used in difference to strlen.